### DISCUSSION TITLE
Adds support for additional parameters when logging in via openid

### DISCUSSION THREAD

**Discussion by valdeirpsr** [directus/directus#18488] [Feature Requests]
Currently, some websites such as Google allow the developer to use various URL parameters in addition to the required ones (client_id, scope, etc).

The modification allows the user to enable a list via `.env` and enables the front-end developer to enhance the integration between the website and Directus.

With Google, for example, the user will be able to choose how the consent screen will be displayed or which pre-registered email of the user will be shown in case of multiple logins. This could facilitate and improve the integration with Directus, since the front-end developer is very limited in terms of the authorization system.

Example code: https://github.com/directus/directus/pull/18478

1. Open `.env`
2. Enter the snippet below
```env
AUTH_PROVIDERS=directus

AUTH_DIRECTUS_DRIVER=openid
AUTH_DIRECTUS_CLIENT_ID=000000000000-abcdefvaldeirpsrbcdefabcdefabcdefab.apps.directususercontent.com
AUTH_DIRECTUS_CLIENT_SECRET=ABCDEF-abcdefghijklmnopqrstuvwxyz
AUTH_DIRECTUS_ISSUER_URL=https://accounts-fake.directus.io
AUTH_DIRECTUS_DEFAULT_ROLE_ID=eddfce8d-36e9-4cba-90a4-6b8b2004a3cc

# Config Here
AUTH_DIRECTUS_CUSTOM_URL_PARAMS=login_hint,display
```
> The example above is just fictitious. Configure with your correct data
3. Go To https://www.your-site.com/auth/login/directus?login_hint=example@directus.com

### Example

**Without login_hint**
![Example without login_hint](https://i.ibb.co/svgjYFs/image.png)

**With login_hint**
![Example with login_hint](https://i.ibb.co/Xbzq7P3/image.png)

As I said before, this simple modification brings an improvement for the front-end, which will have more freedom with the authentication system, mainly when the developer has the email saved on localStorage (or IndexedDB) and needs to validate data and, in case of invalidation, resend the user to the login system.

**Comment by aidenfoxx** [comment]
This already exists in the form of the `PARAMS` config option: https://docs.directus.io/self-hosted/config-options.html#openid

While not well documented, it does accept a JSON object defining the "key:value" params to add to the URL e.g. `AUTH_<PROVIDER>_PARAMS="json:{"prompt":"consent"}"`

**Reply by valdeirpsr** [reply]
How could I use `login_hint` (Google) or `domain_hint` (Azure) with `AUTH_<PROVIDER>_PARAMS`? The existing option only works with static data such as option "prompt".

The proposal is different. `AUTH_<PROVIDER>_PARAMS` is for fixed data. The values need to be defined in the `.env`. The modification I proposed is for dynamic data.

**Reply by aidenfoxx** [reply]
I see what you mean now. There is currently no way to set the params dynamically.

**Comment by valdeirpsr** [comment]
**Modification alternative**

Another way to achieve the same result is by creating a *hook* that would be triggered after generating the authentication URL.
This way, the developer could handle the necessary data before the user is redirected.

**Example**

```javascript
export default ({ filter }) => {
    filter('filter.name', async (input) => {
        /**
        * Logic Here
        */

        return input;
    });
};
```

And

```javascript
export default ({ filter }) => {
    filter('filter.login.error', async (input) => {
        /**
        * Logic Here
        */

        return input;
    });
};
```

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
